docs/specs/native_histograms.md

```diff
@@ -1665,14 +1679,22 @@ retrieved from the TSDB.
 ### Compatibility between histograms
 
 When an operator or function acts on two or more native histograms, the
-histograms involved need to have the same schema and zero bucket width. Within
-certain limits, histograms can be converted on the fly to meet these
-compatibility criteria:
-
-- An NHCB (schema -53) is only ever compatible with other NHCBs that also MUST
-  have the exact same custom values. (In principle, there are possible
-  differences in custom values that could be reconciled, but PromQL doesn't yet
-  consider those.)
+histograms involved need to have the same schema, the same zero bucket width,
+and (if applicable) the same custom values. Within certain limits, histograms
+can be converted on the fly to meet these compatibility criteria:
+
+- NHCBs (schema -53) are only compatible with each other. Different custom
+  values need to be reconciled by conversion in the following way:
+  - Identify the set of buckets with the highest possible number of buckets
```

Comment:
This is a bit too complicated, maybe we can say:

"Take the  custom values from the original NHCBs that occur in both NHCBs (i.e. intersect). These are the new custom values. For example in case of [1, 2, 4, 8, 16] and [1, 4, 16], the new custom values would be [1, 4, 16], thus creating the (-Inf, 1], (1, 4], (4, 16], (16, +Inf] buckets that overlap exactly with the original buckets."